pkg/scheduler/core/generic_scheduler.go

```diff
@@ -651,7 +651,7 @@ func (g *genericScheduler) podFitsOnNode(
 			if err != nil {
 				return false, []predicates.PredicateFailureReason{}, nil, err
 			}
-		} else if !podsAdded || len(failedPredicates) != 0 {
+		} else if !podsAdded || len(failedPredicates) != 0 || !status.IsSuccess() {
```

Comment:
I don't fully understand why we run predicates twice here. But looking at the code, when this branch is reached, the status can only be Success or Unschedulable. So this change is to break if status is Unschedulable?

Reply:
The [comment](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/core/generic_scheduler.go#L626) at the top tries to explain this, do you have questions about that?

Reply:
Now I understand it. Discussed offline, we are going to keep it for now and I am gonna do a followup cleanup.